### ISSUE TITLE
fit_generator get much lower accuracy then the evaluate_generator just one line above? 

### ISSUE THREAD

**Issue by bzhong2** [fchollet/keras#6895]
Hello, 
I am following the blog for transfer learning https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html. 
##First I compute the saved the bottleneck features and build a new model and train it with the bottleneck features: 
input_layer = Input(shape=base_model.output_shape[1:])
x = GlobalAveragePooling2D()(input_layer)
x = Dense(512, activation='relu',name='fc_new_1')(x)
x = Dropout(0.2)(x)
x = Dense(512, activation='relu',name='fc_new_2')(x)
x = Dense(num_classes, activation='softmax',name='logit_new')(x)
Add_layers = Model(inputs=input_layer, outputs=x,name='Add_layers')

##Then I put this new model at the end of pretrained models: 
base_model = ResNet50(include_top=False, weights='imagenet', input_shape=(img_shape[0],img_shape[1],3))
x = base_model.output
predictions = Add_layers(x)
model = Model(inputs=base_model.input, outputs=predictions)
model.compile(optimizer=sgd, loss='categorical_crossentropy', metrics=['accuracy'])

##Then, I evaluate the model : 
score = model.evaluate_generator(train_generator, nb_train_samples // batch_size_finetuning)
print('The evaluation of the entire model before fine tuning : ')
print(score)
score = model.evaluate_generator(validation_generator, nb_validation_samples // batch_size_evaluation)
print('The evaluation of the entire model before fine tuning : ')
print(score)

And get training loss and accuracy : [0.015362062912073827, 1.0]
validation loss and accuracy : [0.89740632474422455, 0.75]
##Just one line below it, I trained the new model: 
model.fit_generator(train_generator,
                    steps_per_epoch= nb_train_samples // batch_size_finetuning,
                    epochs=finetuning_epoch,
                    validation_data=validation_generator,
                    validation_steps=nb_validation_samples // batch_size_evaluation,
                    callbacks=[checkpointer_finetuning,history_finetuning,TB_finetuning,lrate_finetuning,Eartly_Stopping_finetuning]);

Then the output is : 
31/31 [==============================] - 35s - loss: 3.4004 - acc: 0.3297 - val_loss: 0.9591 - val_acc: 0.7083

Weird thing is : This problem only happens if I use Resnet50 and Inceptionv3 but not with vgg16. I am pretty sure that changing the pretrained model is the only difference. I understand that the dropout might make it different but it should not be this large and vgg16 has no obvious problem at all.  

Another weird thing is: If I change every layer to be .trainable = False and compile, the validation accuracy will still decrease dramatically. I even checked the weights of every layer and if .trainable = False weights will not change and .trainable = True weights will change. 

Any help is appreciated!!! THANKS!!!

**Comment by bzhong2** [comment]
I tried to use .fit instead of fit_generator and the same problem is still there: here is the new code:

score = model.evaluate(validation_images, validation_labels, batch_size=batch_size_finetuning)
print('(Validation) Evaluate the entire model : ')
print(score)
### evaluate the added layers
score = model.evaluate(train_images, train_labels, batch_size=batch_size_finetuning)
print('(Train) Evaluate the entire model : ')
print(score)

print('Start Fine tuning...')
history_finetuning = model.fit(train_images, 
						train_labels, 
						epochs=ft_epochs, 
						batch_size=batch_size_finetuning, 
						validation_data=(validation_images, validation_labels),
						shuffle=False);

I then tried the standard data set cifar10 and the results are as follows:
For the first training of the added layers: the last several epochs are as below:
Epoch 397/400
8000/8000 [==============================] - 0s - loss: 0.0012 - acc: 0.9998 - val_loss: 0.4322 - val_acc: 0.9115
Epoch 398/400
8000/8000 [==============================] - 0s - loss: 7.6729e-04 - acc: 1.0000 - val_loss: 0.4322 - val_acc: 0.9095
Epoch 399/400
8000/8000 [==============================] - 0s - loss: 6.1750e-04 - acc: 1.0000 - val_loss: 0.4314 - val_acc: 0.9085
Epoch 400/400
8000/8000 [==============================] - 0s - loss: 0.0014 - acc: 0.9998 - val_loss: 0.4356 - val_acc: 0.9120

Then I evaluate it 
(Validation) Evaluate the entire model :
[0.43556556671857832, 0.9120000194311142]
8000/8000 [==============================] - 34s
(Train) Evaluate the entire model :
[4.2374865099191086e-05, 1.0]

Then I finetune the entire again, it shows: 
Epoch 1/300
  36/8000 [..............................] - ETA: 1393s - loss: 2.4836 - acc: 0.61112017-06-24 10:40:24.645982: I c:\tf_jenkins\home\workspace\release-win\m\windows-gpu\py\35\tensorflow\core\common_runtime\gpu\pool_allocator.cc:247] PoolAllocator: After 1846990 get requests, put_count=1846697 evicted_count=1000 eviction_rate=0.000541507 and unsatisfied allocation rate=0.0007542
 648/8000 [=>............................] - ETA: 149s - loss: 2.1055 - acc: 0.64202017-06-24 10:40:31.522072: I c:\tf_jenkins\home\workspace\release-win\m\windows-gpu\py\35\tensorflow\core\common_runtime\gpu\pool_allocator.cc:247] PoolAllocator: After 2377 get requests, put_count=2300 evicted_count=1000 eviction_rate=0.434783 and unsatisfied allocation rate=0.460244
2017-06-24 10:40:31.522508: I c:\tf_jenkins\home\workspace\release-win\m\windows-gpu\py\35\tensorflow\core\common_runtime\gpu\pool_allocator.cc:259] Raising pool_size_limit_ from 193 to 212
1584/8000 [====>.........................] - ETA: 96s - loss: 1.9212 - acc: 0.65782017-06-24 10:40:42.080824: I c:\tf_jenkins\home\workspace\release-win\m\windows-gpu\py\35\tensorflow\core\common_runtime\gpu\pool_allocator.cc:247] PoolAllocator: After 4773 get requests, put_count=4828 evicted_count=1000 eviction_rate=0.207125 and unsatisfied allocation rate=0.207207
2017-06-24 10:40:42.081293: I c:\tf_jenkins\home\workspace\release-win\m\windows-gpu\py\35\tensorflow\core\common_runtime\gpu\pool_allocator.cc:259] Raising pool_size_limit_ from 493 to 542
8000/8000 [==============================] - 104s - loss: 1.2794 - acc: 0.7563 - val_loss: 0.7371 - val_acc: 0.8465
Epoch 2/300
8000/8000 [==============================] - 99s - loss: 0.6549 - acc: 0.8411 - val_loss: 0.8990 - val_acc: 0.8265

Is it normal because of dropout? But why VGG16 will not be like this but only inception3 and resnet50 and the difference is so large?

**Comment by bzhong2** [comment]
@ouzan19 @danielS91 @fchollet Do you think it might because of similar reasons as #6977?

**Comment by bzhong2** [comment]
I checked it for many times. I am pretty sure that the code is the same but only the based model is vgg16 then everything is expected, the second training starts with the accuracy as the evaluation and the end of the first training. But Resnet50 and inception3 will all experience a significant drop from the evaluation to the beginning of the second training. The evaluation is the same as the end of first training, so I assume that attaching the new sequential model (Added_layers) to the base models (inception3 or resnet50) is not wrong, otherwise the evaluation should not give expected answer. Really confused about what happened.

**Comment by renato145** [comment]
This is also happening to me.
Wierd thing: if you load the base_model, put your top layers and evaluate the model, it behaves as expected, but as soon as you train it (even when freezing the layers and using a very small lr) it breaks, and the evaluation will drop by a lot.

**Comment by Superlee506** [comment]
@bzhong2 @renato145 
This problem happens to me, it confuses me a lot.
Do you find a solution?

**Comment by Fuze5k** [comment]
answer here
https://github.com/tensorflow/tensorflow/issues/44568
